tests/unit/s2n_self_talk_client_hello_cb_test.c

```diff
@@ -434,6 +479,37 @@ int run_test_reject_handshake_ch_cb(s2n_client_hello_cb_mode cb_mode,
     return S2N_SUCCESS;
 }
 
+int run_test_poll_ch_cb(s2n_client_hello_cb_mode cb_mode,
```

Comment:
Can you make sure you test this with a hello retry handshake?

Reply:
done in https://github.com/aws/s2n-tls/pull/3230/commits/e33558485fdee584cf9737bd88a6d2c949aa1516 and https://github.com/aws/s2n-tls/pull/3230/commits/1bb2f7033e809d0704cb2875d4224b68b72723aa